docs/sidechain/spec.md

```diff
@@ -0,0 +1,945 @@
+
+A cross-chain transfer, as described in the section above, requires an account
+on the destination chain to checkout a "cross-chain sequence number". This makes
+it difficult to create new accounts using cross-chain transfers. A dedicated
+transaction is used to create accounts: `SidechainXChainAccountCreate`. This
+specifies the same information as a `SidechainXChainTransfer`, but the
+destination account is not longer optional, and the amount is in XRP. The XRP
+amount must be greater than or equal to the min creation amount specified in the
+sidechain ledger object. If this optional amount is not present, the transaction
+will fail. Once this transaction is submitted, it works similarly to a
+cross-chain transfer, except the signatures are collected on a
+`XChainAccountCreateState` ledger object on the door account. If the
+account already exists, the transaction will fail. Accounts created this must be
+undeletable or the the signatures could be resubmitted to collect the funds more
+than once.
+
+The `SidechainAddWitness` transaction is used to by witness servers (or accounts
+that use witness servers) to add a witness's attestation that some event
+happened.
+
+There is also a `SidechainModify` transaction used to change new account
+creation amounts and the reward amounts. Note: if the reward amount changes
+between the time a transaction is initiated the time the reward is collected,
+the old amount is used (as that is the amount the source account paid).
+
+
+## Witness Server
+
+A witness server is an independent server that helps provide proof that some
+event happened on either the mainchain or the sidechain. When they detect an
+event of interest, they use the `SidechainAddWitness` transaction to add their
+attestation that the event happened. When a quorum on signatures are collected on
+the ledger, the transaction predicated on that event happening is unlocked (and
+may be triggered automatically when the quorum of signatures is reached).
+Witness servers are independent from the servers that run the chains themselves.
+
+It is possible for a witness server to provide attestations for one chain only -
+and it is possible for the door account on the mainchain to have a different
+signer's list than the door account on the sidechain. The initial implementation
+of the witness server assumes it is providing attestation for both chains,
+however it is desirable to allow witness servers that only know about one of
+the chains.
+
+The current design envisions two models for how witness servers are used. In the
+first model, the servers are completely private. They submit transactions to the
+chains themselves and collect the rewards themselves. Allowing the servers to be
+private had the advantage of greatly reducing the attack surface on these
+servers. They won't have to deal with adversarial input to their RPC commands,
+and since their ip address will be unknown, it will be hard to mount an DOS
```

Comment:
dumb question but is it possible to guess their IP address by looking at rippled logson mainchain or sidechain or trx data on the ledger explorer?

Reply:
The logs are private, and you can't guess the ip of the rippled instance that submitted the transaction from the transaction itself (and even if you could, that would be the ip of the ripple server, not the witness server)

Reply:
sounds good